datafusion/core/src/execution/context/mod.rs

```diff
@@ -2047,6 +2052,11 @@ impl<'a> ContextProvider for SessionContextProvider<'a> {
     }
 
     fn get_function_meta(&self, name: &str) -> Option<Arc<ScalarUDF>> {
+        #[cfg(feature = "array_expressions")]
+        if let Some(udf) = functions_array::get_array_udf(name) {
+            return Some(udf);
+        }
+
         self.state.scalar_functions().get(name).cloned()
```

Comment:
fallback to user defiend function. Assume they can register actual function . If they need to register closure instead, call `array_functions().lock().unwrap().insert(name.to_string(), udf)`